Reverting TIOCPKT changes. Not suitable for upstream.
Without this patch, gnome-terminal was not functioning as expected 
and following issues were seen:
1. Alignment(whitespace) issue.
2. Username sometimes doesn’t display the first character(“dmin” instead of “admin”). This issue maybe a result of the above mentioned alignment issue.
3. Also, whatever commands are typed weren’t being displayed and only output of the command were displayed.

--- a/src/vte.cc	2021-04-23 16:04:32.914432936 -0700
+++ b/src/vte.cc	2021-04-23 16:09:35.901327996 -0700
@@ -3964,12 +3964,16 @@
                                  * We need to see what that byte is, but otherwise drop it
                                  * and write continuously to chunk->data.
                                  */
+#if 0
                                 char pkt_header;
                                 char save = bp[-1];
+#endif
                                 errno = 0;
-                                int ret = read (fd, bp - 1, rem + 1);
+                                int ret = read (fd, bp, rem);
+#if 0
                                 pkt_header = bp[-1];
                                 bp[-1] = save;
+#endif
 				switch (ret){
 					case -1:
 						err = errno;
@@ -3978,6 +3982,7 @@
 						eos = true;
 						goto out;
 					default:
+#if 0
                                                 ret--;
 
                                                 if (pkt_header == TIOCPKT_DATA) {
@@ -4005,6 +4010,10 @@
                                                                 pty_scroll_lock_changed(false);
                                                         }
                                                 }
+#endif
+						bp += ret;
+                                                rem -= ret;
+                                                len += ret;
 						break;
 				}
 			} while (rem);
--- a/src/pty.cc	2021-04-23 16:09:55.607590082 -0700
+++ b/src/pty.cc	2021-04-23 16:12:13.160149346 -0700
@@ -353,10 +353,20 @@
 static int
 fd_set_cpkt(vte::libc::FD& fd)
 {
-        /* tty_ioctl(4) -> every read() gives an extra byte at the beginning
-         * notifying us of stop/start (^S/^Q) events. */
-        int one = 1;
-        return ioctl(fd.get(), TIOCPKT, &one);
+        int flags = fcntl(fd.get(), F_GETFL, 0);
+        if (flags >= 0 &&
+            (flags & O_NONBLOCK) == 0)
+                flags = fcntl(fd.get(), F_SETFL, flags | O_NONBLOCK);
+
+        if (flags < 0) {
+                int errsv = errno; GError **error=NULL;
+                g_set_error(error, VTE_PTY_ERROR,
+                            VTE_PTY_ERROR_PTY98_FAILED,
+                            "%s failed: %s", "Setting O_NONBLOCK flag", g_strerror(errsv));
+                errno = errsv;
+                return FALSE;
+        }
+        return TRUE;
 }
 
 static int
